lib/components/steps/Home.tsx

```diff
@@ -1,17 +1,38 @@
+import React from 'react';
 import { Box, Button, Typography, Chip, TableContainer, Paper } from '@mui/material';
 import { StyledTable } from '../common/StyledTable.tsx';
 import { CircleChipDefault, CircleChipSuccess } from '../../icons/index.tsx';
 import { vars } from '../../theme/variables.ts';
 import { STEPS } from "../../models.ts";
 import { useCdeContext } from "../../CdeContext.ts";
+import { isRowMapped } from '../../helpers/getters.ts';
 
 const { primary600, gray600, drodownDetailBg, gray200 } = vars;
 
 function Home() {
     const {
         setStep,
-        datasetSample
+        datasetSample,
+        name,
+        datasetMapping,
+        headerIndexes,
+        getSuggestions
     } = useCdeContext();
+    
+    const [numberOfMapped, setNumberOfMapped] = React.useState(0);
+    const suggestionsMapping = getSuggestions();
+
+    const sumOfMappedAndUnmapped = Object.keys(datasetMapping).length;
+    const numberOfUnmapped = sumOfMappedAndUnmapped - numberOfMapped;
```

Comment:
remark: @Aiga115 I would do the computation only at line 76 and remove the variable here

Reply:
I see, will fix that @zsinnema

Reply:
@zsinnema please have a look at the changes done in commit [1d6e643](https://github.com/MetaCell/cde-mapper/pull/26/commits/1d6e643b98e6f1a24c1fa395c8ad714a310ed84e). Based on our conversation, I decided to split the cde context into 3 different contexts with different purposes (data, services and ui).
- Data Context: Holds the application's core data;
- Service Context: Encapsulates functions and operations that act upon the data;
- UI Context: Manages UI-related state and interaction;

For the components that we will create, if they need the data exactly as it is stored, I believe it's fine to use the data context; However, if the component needs to perform some transformation or do some calculations based on the data stored it should use the service context.